File: src/Domain/Entities/TratamientoVeterinario.cs
using TrazabilidadIberica.Domain.Common;

namespace TrazabilidadIberica.Domain.Entities;

public class TratamientoVeterinario : BaseEntity
{
    public Guid AnimalId { get; set; }
    public Guid VeterinarioId { get; set; }
    public required string NombreMedicamento { get; set; }
    public string? NumeroLote { get; set; }
    public DateTime FechaAdministracion { get; set; }
    public DateTime? FechaCaducidad { get; set; }
    public decimal DosisAdministrada { get; set; }
    public string? UnidadDosis { get; set; }
    public string? ViaAdministracion { get; set; }
    public int PeriodoSupresionDias { get; set; }
    public DateTime FechaFinSupresion { get; set; }

    public Animal Animal { get; set; } = null!;
    public Veterinario Veterinario { get; set; } = null!;
}

File: frontend/src/app/core/models/tratamiento-veterinario.ts
export interface TratamientoVeterinario {
  id: string;
  animalId: string;
  veterinarioId: string;
  nombreMedicamento: string;
  numeroLote: string | null;
  fechaAdministracion: string;
  fechaCaducidad: string | null;
  dosisAdministrada: number;
  unidadDosis: string | null;
  viaAdministracion: string | null;
  periodoSupresionDias: number;
  fechaFinSupresion: string;
  animal?: { id: string; numeroCrotal: string };
  veterinario?: { id: string; nombreCompleto: string };
  createdOffline: boolean;
  clientId: string | null;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
}
